src/Solvers/fft_based_poisson_solver.jl

```diff
@@ -32,38 +55,47 @@ function FFTBasedPoissonSolver(arch, grid, planner_flag=FFTW.PATIENT)
 end
 
 """
-    solve_poisson_equation!(solver)
+    solve!(x, poisson_solver, b, r=0)
 
-Solves Poisson's equation ∇²ϕ = RHS where `RHS = solver.storage` using periodic or staggered
-Neumann boundary conditions as determined by the `solver.grid`. `solver.storage` will be mutated
-to contain the solution.
+Solves the "screened" poisson equation
```

Comment:
```suggestion
Solves the "screened" Poisson equation
```

Reply:
What is `m`? Anything?
Perhaps semantics but in general I think "screened Poisson eq" refers to (∇² - m²) ϕ = b. The (∇² + m²) ϕ = b is called "Helmholtz eq".

Reply:
Here `m` can be positive or negative as far as I can tell. Our application uses the screened Poisson equation. The Helmholtz equation is an eigenvalue problem I guess, so this solver isn't generally applicable to that case. We could call the constant m^2. I don't know if that's helpful though.

Reply:
No I didn't mean we change m to m^2...

But why you say that Helmholtz equation is eigenvalue problem?

https://en.wikipedia.org/wiki/Screened_Poisson_equation
https://en.wikipedia.org/wiki/Helmholtz_equation#Inhomogeneous_Helmholtz_equation

Reply:
Err... sorry, I guess what I meant is that there are some RHS that are resonant. But you're right, we can solve it.

So should we say that we can "either solve the screened Poisson equation or Helmholtz equation"?

Reply:
I think we don't even need to call it anything. Just write,
```Julia
Solves the equation,
```math
...
```
```

Reply:
Sounds good.

Reply:
Bah, the issue here is that the type is `FFTBasedPoissonSolver`. Is "screened" Poisson really confusing?

Reply:
I just mean its like Poisson but with a constant, limits to Poisson when `m=0`. What's the right way to describe it?

Reply:
Keep it! It's ok!! :)

Reply:
Let me know what you think now.